src/libs/safe-path/README.md

```diff
@@ -0,0 +1,43 @@
+Safe Path
+====================
+[![CI](https://github.com/magiclen/path-absolutize/actions/workflows/ci.yml/badge.svg)](https://github.com/magiclen/path-absolutize/actions/workflows/ci.yml)
+
+A library to safely handle filesystem paths, typically for container runtimes.
+
+Linux [mount namespace](https://man7.org/linux/man-pages/man7/mount_namespaces.7.html) provides isolation of the list
+of mounts seen by the processes in each [namespace](https://man7.org/linux/man-pages/man7/namespaces.7.html) instance.
+Thus, the processes in each of the mount namespace instances will see distinct single-directory hierarchies.
+
+Containers are used to isolate workloads from the host system. Container on Linux systems depends on the mount namespace
+to build an isolated root filesystem for each container, thus protect the host and containers from each other.
+When creating a container, the container runtime needs to setup filesystem mounts for the container according to
+configuration information from:
+- the container image
+- K8S pod specification
+- command line arguments
+
+These volume configuration information may be controlled by end users/malicious attackers, so it must not be trusted
+by container runtimes. When the container runtime is preparing mount namespace for a container, it must be very careful
+to validate user input configuration information and ensure data out of the container rootfs directory won't be affected
+by the container. There are several types of attacks related to container mount namespace:
+- symlink based attack
+- Time of check to time of use(TOCTTOU)
+
+The `safe-path` crate provides several mechanisms for container runtimes to safely handle filesystem paths when
+preparing mount namespace for containers.
+- [safe_join]: safely joins two given paths and the returned path is guaranteed to be scoped inside `root`.
```

Comment:
safe_join -> `safe_join` .

And the `[]` can be deleted I think.

Reply:
Add current doc links for them